### PR TITLE
Update integration after recent deprecations

### PR DESCRIPTION
**Pull request by AleksanderWWW**
Will add tests in a seperate PR

### CODE REVIEW

**Review by normandy7 — APPROVED** [review]

File: src/neptune_sacred/impl/__init__.py [hunk 1]
```diff
@@ -17,47 +17,47 @@
 import json
 import os
 import warnings
+from typing import Union
 
+from neptune.new.handler import Handler
```
Comment (kshitij12345) [inline src/neptune_sacred/impl/__init__.py]: Wouldn't this lead to user warning with 1.0 release?

Shouldn't we keep it in the `try` block?

File: src/neptune_sacred/impl/__init__.py [hunk 2]
```diff
@@ -72,38 +72,50 @@ class NeptuneObserver(RunObserver):
...
+    You may also want to check the following:
 
-    .. _sacred integration docs page:
-        https://docs.neptune.ai/integrations-and-supported-tools/model-training/sacred
-    .. _example experiment page:
-        https://app.neptune.ai/prince.canuma/sacred-integration/e/SAC-59/all
+    Sacred integration docs page:
+        https://docs.neptune.ai/integrations/sacred
+    Example run displayed in Neptune:
+        https://app.neptune.ai/o/common/org/sacred-integration/e/SAC-1336/all
```
Comment (normandy7) [inline src/neptune_sacred/impl/__init__.py]: The only issue I see here is that the example dashboard includes hardware consumption metrics, but they didn't seem to be logged for this run.

Not sure how important it is, but we probably want to either capture those or alternatively remove those widgets from the dashboard.

Comment (AleksanderWWW) [reply src/neptune_sacred/impl/__init__.py]: @normandy7 updated

File: src/neptune_sacred/impl/__init__.py (hunk 1 above)
Comment (twolodzko) [reply src/neptune_sacred/impl/__init__.py]: We should.

**Review by twolodzko — COMMENTED** [review]
Don't we need to use `stringify_unsupported` here as well https://github.com/neptune-ai/neptune-sacred/pull/20/files#diff-b594055c60dd2fcab8521b68c27079ef8f1e1c64282b3b330dd38b454248a4cfR129-R130 ?

**Comment by AleksanderWWW** [comment]
> Don't we need to use `stringify_unsupported` here as well https://github.com/neptune-ai/neptune-sacred/pull/20/files#diff-b594055c60dd2fcab8521b68c27079ef8f1e1c64282b3b330dd38b454248a4cfR129-R130 ?

I don't think so tbh - the only thing that would require it is when there is a dict type, but what is logged is not the dict itself, but rather the JSON string (obtained from `json.dumps`). So I guess it is a supported type, is it not?

File: src/neptune_sacred/impl/__init__.py [hunk 3]
```diff
@@ -13,51 +13,61 @@
...
+    from neptune.handler import Handler
+    from neptune.types import File
+    from neptune.utils import stringify_unsupported
+except ImportError:
+    # neptune-client=0.9.0+ package structure
+    from neptune.new.types import File
+    from neptune.new.handler import Handler
+    from neptune.new.metadata_containers import Run
+    from neptune.new.utils import stringify_unsupported
+
```
Comment (twolodzko) [inline src/neptune_sacred/impl/__init__.py]: to update

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,8 @@
+## [UNRELEASED] neptune-sacred 0.11.0
```
Comment (kshitij12345) [inline CHANGELOG.md]: Should this be a major release for integration i.e. 1.0?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,8 @@
...
+
+### Changes
+- Updated integration to be compatible with `neptune-client` `1.0.0`. ([#20](https://github.com/neptune-ai/neptune-sacred/pull/20))
```
Comment (Raalsky) [inline CHANGELOG.md]: `neptune` (instead of `neptune-client`)?

File: tests/test_e2e.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import neptune.new as neptune
+import torch
+from neptune.new.integrations.sacred import NeptuneObserver
+
+
+def test_e2e(experiment, model):
+    if torch.device("cuda:0"):
```
Comment (Raalsky) [inline tests/test_e2e.py]: Why do we need it? We're not running it on GPU on CI/CD. Is it for local testing?

File: src/neptune_sacred/impl/__init__.py [hunk 7]
```diff
@@ -72,38 +73,50 @@ class NeptuneObserver(RunObserver):
...
+    Sacred integration docs page:
+        https://docs.neptune.ai/integrations/sacred
+    Example run displayed in Neptune:
+        https://app.neptune.ai/o/common/org/sacred-integration/e/SAC-1341/all
     """
 
-    def __init__(self, run, base_namespace="experiment"):
+    def __init__(
+        self,
+        run: Run | Handler,
```
Comment (Raalsky) [inline src/neptune_sacred/impl/__init__.py:81]: How about testing it with `Handler`?

File: src/neptune_sacred/impl/__init__.py [hunk 8]
```diff
@@ -72,38 +73,50 @@ class NeptuneObserver(RunObserver):
...
+    ):
+
         super(NeptuneObserver, self).__init__()
-        expect_not_an_experiment(run)
+
         self._run = run
 
+        if isinstance(self._run, Handler):
+            self._root_object = self._run.get_root_object()
+        else:
```
Comment (Raalsky) [inline src/neptune_sacred/impl/__init__.py:93]: Let's still have `expect_not_an_experiment(run)` here

File: tests/test_e2e.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
+import neptune.new as neptune
```
Comment (kshitij12345) [inline tests/test_e2e.py]: `import neptune`

File: tests/test_e2e.py [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+import torch
+from neptune.new.integrations.sacred import NeptuneObserver
```
Comment (kshitij12345) [inline tests/test_e2e.py]: `from neptune.integrations.sacred import NeptuneObserver`

**Review by kshitij12345 — CHANGES_REQUESTED** [review]
minor suggestions and one question. Thank you

**Review by kshitij12345 — CHANGES_REQUESTED** [review]
minor suggestions and one question. Thank you

File: .github/workflows/ci.yml
Comment (twolodzko) [inline .github/workflows/ci.yml:1]: Did you also send a PR to the client repo to add it to the e2e tests?

Comment (AleksanderWWW) [reply .github/workflows/ci.yml:1]: Not yet, will do it after the merge here, so I can configure the tests on client to be taken from the master branch

File: src/neptune_sacred/impl/__init__.py [hunk 11]
```diff
@@ -13,97 +13,102 @@
...
 import warnings
 
+try:
+    from neptune import Run
+    from neptune.handler import Handler
+    from neptune.integrations.utils import expect_not_an_experiment
+    from neptune.types import File
+    from neptune.utils import stringify_unsupported
+except ImportError:
+    from neptune.new.metadata_containers import Run
```
Comment (kshitij12345) [inline src/neptune_sacred/impl/__init__.py]: from neptune.new import Run?

File: src/neptune_sacred/impl/__init__.py [hunk 12]
```diff
@@ -114,7 +119,7 @@ def completed_event(self, stop_time, result: dict):
                     self._run[self.base_namespace][f"metrics/results/{k}"] = float(v)
                 elif isinstance(v, list) or isinstance(v, dict):
                     self._run[self.base_namespace][f"metrics/results/{k}"] = json.dumps(v)
-                elif isinstance(v, object):
+                elif isinstance(v, File):
```
Comment (kshitij12345) [inline src/neptune_sacred/impl/__init__.py:122]: Would it make sense to add it to CHANGELOG that now we only log few selected types. (object would probably be catch-all, though I doubt that we logged anything meaningful.)

File: tests/test_e2e.py [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+
+    run = neptune_handler.get_root_object()
+    run.sync()
+
+    assert run["sacred/experiment/config/data_dir"].fetch() == "data/CIFAR10"
+
+    assert run.exists("sacred/experiment/io_files/artifacts/model")
+    assert run.exists("sacred/experiment/io_files/artifacts/model_arch")
+
+    assert 0 < run["sacred/experiment/metrics/results/final_acc"].fetch() < 1
```
Comment (kshitij12345) [inline tests/test_e2e.py]: 0 <= acc <= 1?

File: tests/conftest.py [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize([0.485, 0.456, 0.406], [0.229, 0.224, 0.225]),
+                ]
+            )
+        }
+        lr = 1e-2  # noqa: F841
+        bs = 128  # noqa: F841
+        n_classes = 10  # noqa: F841
+        device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")  # noqa: F841
```
Comment (kshitij12345) [inline tests/conftest.py:46]: Like @Raalsky mentioned, would it make sense to simplify code by hard-coding on `cpu` as CI doesn't have GPU yet.

**Review by kshitij12345 — APPROVED** [review]
One minor question (non-blocking)

LGTM! Thanks @AleksanderWWW

File: tests/conftest.py (hunk 14 above)
Comment (AleksanderWWW) [reply tests/conftest.py:46]: Yes, but I think that users could also run the tests themselves for some reasons and they might have a gpu. We could discuss it seperately

**Pull request merged**